### ISSUE TITLE
Handling malformed HTML

### ISSUE THREAD

**Issue by KaneTW** [mlcfp/zenacy-html#1]
I'm parsing a website where users can post a limited subset of HTML. The HTML is always within e.g. a `<div class="user">`.

Users often post malformed HTML, which breaks the parse (e.g. I expect `<div id="potatoes">` as a sibling of `<div class="user">`, but due to malformed tags it's no longer a sibling)

Is there a way to just treat everything within a `<div class="user">` as unparsed text?

**Comment by KaneTW** [comment]
I've investigated a bit more and here's an example (not fully minimal, but showcases the issue):
```
<html>
	<head></head>
	<body>
		<div id="abc">
			<div class="a b">
				<div class="a b">
					<blockquote>
						<p>
							<i>a b c
								<i></i>
							</i>
						</p>
						<p>
							<i>
								<i>a 
									<i> b 
										<i></i>
									</i>
								</i>
							</i>
						</p>
					</blockquote>
				</div>
			</div>
			<div id="cdf"></div>
		</div>
	</body>
</html>
```

Output example:
```
map (htmlElemName &&& htmlElementAttributes) $ filter htmlNodeIsElem  $ htmlElemContent body

[("div",[HTMLAttr {htmlAttrName = "id", htmlAttrVal = "abc", htmlAttrNamespace = HTMLAttrNamespaceNone}]),("i",[])]
```

You can see how the i tag is messed up, and that breaks a lot of the structure.

**Comment by mlcfp** [comment]
Thanks for reporting this issue.  I've reduced the example further to:

```
<html><body> <p><i><i>a<i>b<i></i></i></i></i></p> </body></html>
```

I'm investigating and will report back soon.

**Comment by mlcfp** [comment]
This issue was a bug that is now fixed in 2.1.0
